src/submission/models.py

```diff
@@ -1240,3 +1242,17 @@ def handle_defaults(self, article):
             article.license = self.default_license
 
         article.save()
+
+
+# Signals
+
+@receiver(pre_delete, sender=FrozenAuthor)
+def remove_author_from_article(sender, instance, **kwargs):
+    """
+    This signal will remove an author from a paper if the user deletes the
+    frozen author record to ensure they are in sync.
+    :param sender: FrozenAuthor class
+    :param instance: FrozenAuthor instance
+    :return: None
+    """
+    instance.article.authors.remove(instance.author)
```

Comment:
do we not need to remove the ArticleAuthorOrder record here as well? (shame that there is no signal for removing m2m)

Reply:
We do, commit added.